src/pages/index.tsx

```diff
@@ -31,8 +30,40 @@ const Home: NextPage = () => {
   );
 };
 
-Home.getInitialProps = () => {
-  return { isHome: true };
+const prefetchData = async (networkUrl: string): Promise<QueryClient> => {
+  console.log('prefetching...', networkUrl);
+  const queryClient = new QueryClient();
+  const prefetchOptions = { staleTime: 5000 };
+  const queries = getHomeQueries(networkUrl);
+  await Promise.all([
+    queryClient.prefetchInfiniteQuery(
+      ['blocks'],
+      queries.fetchBlocks(DEFAULT_BLOCKS_LIST_LIMIT, 0),
+      prefetchOptions
+    ),
+    queryClient.prefetchInfiniteQuery(
+      ['confirmedTransactions'],
+      queries.fetchConfirmedTransactions(DEFAULT_LIST_LIMIT_SMALL, 0),
+      prefetchOptions
+    ),
+    queryClient.prefetchInfiniteQuery(
+      ['mempoolTransactions'],
+      queries.fetchMempoolTransactions(DEFAULT_LIST_LIMIT_SMALL, 0),
+      prefetchOptions
+    ),
+  ]);
+  console.log('prefetching done!');
+  return queryClient;
 };
 
-export default withInitialQueries(Home, pageAtomBuilders)(getHomePageQueries);
+export const getServerSideProps = wrapper.getServerSideProps(store => async ({ query }) => {
+  const client = await prefetchData(store.getState().global.activeNetwork.url);
+  return {
+    props: {
+      isHome: true,
+      dehydratedState: JSON.parse(JSON.stringify(dehydrate(client))),
```

Comment:
would be helpful to extract this to an extra method with a name explaining how you parse `undefined` values